android/src/main/java/com/bugsnag/BugsnagReactNative.java

```diff
@@ -142,10 +138,11 @@ public void notifyBlocking(ReadableMap payload, boolean blocking, com.facebook.r
       map.put("severity", severity);
       map.put("severityReason", severityReason);
 
-      Bugsnag.internalClientNotify(exc, map, blocking, handler);
+      //always block 
+      Bugsnag.internalClientNotify(exc, map, true, handler);
```

Comment:
Right, though if you call `notify()` from your own code (instead of from the exception handler), that should be a quick non-blocking operation. That's why there was logic to switch between the two.

Reply:
I get that, but in the case of a quick `notify` call (non exception), the bugsnag JS SDK does not do anything with the result of that call (does not perform any action in the `then` callback of the promise). 

If there was an action to take in the javascript SDK after a call to `notify` then you would be correct that we should not make that blocking... But because no action needs to be taken, it should matter if the native SDK is using the blocking API or not in this case.

In the case of an unhandled exception, that is when we do care about the callback/promise, so it should be blocking.

Reply:
Ah, I see what you're saying. You're right, it doesn't block the JS thread. My concern was that it would block the native thread called from JS, in case that needed to be reused or would block other JS-to-native operations.

Reply:
Ah yes, your right. We don't want to hold up the JS execution thread if we don't have to (in the case of a normal handled call to `notify`. 

It would be nice to keep the unified API cross platform. Perhaps we can change my existing `notify` to simply include a boolean? iOS can just ignore it?

Reply:
Yeah, that's what I went with! Extended this PR in #290.

Reply:
Awesome! Thanks for fixing this in a timely manner :)